src/itf/transition-information.h

```diff
@@ -38,27 +38,69 @@ namespace kaldi {
 class TransitionInformation {
  public:
   virtual ~TransitionInformation() {};
-  virtual int32_t TransitionIdToTransitionState(int32_t trans_id) const = 0;
   /**
-   * Phone should really be Token here, where Token could be a word
-   * piece, phone, or character. However, because the original
-   * class, TransitionModel, used Phone, we retain it for the sake
-   * of not changing all the callers of this method.
+   * Returns true if trans_id1 and trans_id2 can correspond to the
+   * same phone when trans_id1 immediately precedes trans_id2 (i.e.,
+   * trans_id1 occurss at timestep t, and trans_id2 ocurs at timestep
+   * 2) (possibly with epsilons between trans_id1 and trans_id2) OR
+   * trans_id1 ocurs before trans_id2, with some number of
+   * trans_id_{k} values, all of which fulfill
+   * TransitionIdsEquivalent(trans_id1, trans_id_{k})
+   *
+   * If trans_id1 == trans_id2, it must be the case that
+   * TransitionIdsEquivalent(trans_id1, trans_id2) == true
+   */
+  virtual bool TransitionIdsEquivalent(int32_t trans_id1, int32_t trans_id2) const = 0;
+  /**
+   * Returns true if this trans_id corresponds to the start of a
+   * phone.
+   */
+  virtual bool TransitionIdIsStartOfToken(int32_t trans_id) const = 0;
+  /**
+   * Phone is understood to correspond not only to linguistic phones,
+   * but also graphemes, word pieces, byte pieces, etc. A better word
+   * would be Token. We use the word Phone for backwards compatibility
+   * with TransitionModel, which was designed only with phones in
+   * mind. Since TransitionInformation was added to subsume
+   * TransitionModel, we did not want to change the call site of every
+   * TransitionModel::TransitionIdToPhone to
+   * TransitionInformation::TransitionIdToToken.
    */
   virtual int32_t TransitionIdToPhone(int32_t trans_id) const = 0;
+  /**
```

Comment:
I encourage you to leave the `Phone` alone. It does not matter _what_ you call it, the user adapts to terminology. It's not linguistically a phone anyway. Most if not all lexicons (TIMIT is likely the only exception) are phonemic (cat = /kat/, kitten = /ˈkitən/, butter = /ˈbətər/), not phonetic (cat = [kʰæʔ], [kʰæt̚], kitten = ['kʰıʔn̩], butter = ['bʌɾɚ], ['bʌɾɹ̩]). So it's properly a `Phoneme`, not `Phone`. I fully subscribe to the de Saussure's notion that « le lien unifiant le signifiant et le signifié est arbitraire [ in a more precise sense, _immotivé_, as he remarks further — kkm] ». In other words, I give no fuzz about it. 

But what happened now, `Phone` is still in `TransitionIdToPhone`, but the previous method calls the same thing `Token`. If you _strongly_ feel that `Phone` is _very_ misleading and must be renamed `Token`, it should be consistentl. This will have a fallout among the existing Kaldi user base, so please take that into a consideration. I'd leave it alone just for convenience of leaving it alone. :) The Doxy comment at L55 still calls it a phone...

I'd also rather omit the long excuse/explanation why "Phone". The documentation should be concise and precise. Something like
```
   * Phone is a historical term, and should be understood in a wider sense that
   * also includes graphemes, word pieces, etc.: any minimal entity in your
   * problem domain which is represented by a sequence of transitions with
   * a PDF assigned to each of them by the model.
```
"Historical" says it all without a lengthy explanation that reads like "we're sorry, we did not think about it." This is not very informative.

I can live with the precise linguistic term "word piece." :) No problem here.

I understand you don't want to change all `TransitionIdToPhone` to `TransitionIdToToken` right now because of the sheer scope of the change. My preference is leave the thing called `Phone`. If you strongly feel it has to be changed, let's commit this one and then change it. But as is, the presence of both `Token` and `Phone` in the same interface and with the same semantics is very confusing (especially given that TM is pretty loaded with intimately interconnected things: phone, state ID, transition ID, PDF ID), and I'm not really excited by adding a synonym to one of the terms in this set.

Reply:
Phone it is. I value backwards compatibility too much to consider changing to Token.

Thanks for pointing out that `TransitionIdIsStartOfToken` should be consistent with TransitionIdToPhone.

Reply:
Cool! So I am merging this as soon as Travis passes? We build there with CUDA off, let's see if anything pops out.